repository/src/test/java/org/alfresco/repo/action/executer/NodeSizeActionExecuterTest.java

```diff
@@ -0,0 +1,109 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.repo.action.executer;
+
+import org.alfresco.model.ContentModel;
+import org.alfresco.model.FolderSizeModel;
+import org.alfresco.repo.action.ActionImpl;
+import org.alfresco.repo.security.authentication.AuthenticationComponent;
+import org.alfresco.service.cmr.repository.NodeRef;
+import org.alfresco.service.cmr.repository.NodeService;
+import org.alfresco.service.cmr.repository.StoreRef;
+import org.alfresco.service.namespace.QName;
+import org.alfresco.util.BaseSpringTest;
+import org.alfresco.util.GUID;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.transaction.annotation.Transactional;
+
+@Transactional
+public class NodeSizeActionExecuterTest extends BaseSpringTest
+{
+    /**
+     * The node service
+     */
+    private NodeService nodeService;
+
+    /**
+     * The test node reference
+     */
+    private NodeRef nodeRef;
+
+    /**
+     * The folder Size executer
+     */
+    private NodeSizeActionExecuter executer;
+
+    /**
+     * Id used to identify the test action created
+     */
+    private final static String ID = GUID.generate();
+
+    /**
+     * Called at the begining of all tests.
+     */
+    @Before
+    public void before() throws Exception
+    {
+
+        this.nodeService = (NodeService)this.applicationContext.getBean("nodeService");
+        StoreRef testStoreRef;
+        NodeRef rootNodeRef;
+
+        AuthenticationComponent authenticationComponent = (AuthenticationComponent)applicationContext.getBean("authenticationComponent");
+        authenticationComponent.setCurrentUser(authenticationComponent.getSystemUserName());
+
+        // Create the store and get the root node
+        testStoreRef = this.nodeService.createStore(StoreRef.PROTOCOL_WORKSPACE, "Test_"
+                        + System.currentTimeMillis());
+        rootNodeRef = this.nodeService.getRootNode(testStoreRef);
+
+        // Create the node used for tests
+        this.nodeRef = this.nodeService.createNode(
+                rootNodeRef,
+                ContentModel.ASSOC_CHILDREN,
+                QName.createQName("{test}testnode"),
+                ContentModel.TYPE_CONTENT).getChildRef();
+
+        // Get the executer instance.
+        this.executer = (NodeSizeActionExecuter)this.applicationContext.getBean(NodeSizeActionExecuter.NAME);
+    }
+
+    /**
+     * Test execution.
+     */
+    @Test
```

Comment:
Add 2 Test cases
1. One for performance test where we add > 200 nodes with nested folders and files
2. Fetching the records with Pagination.

Reply:
3. Use the Alfresco coding formatter to format the code.

Reply:
Done.